tactical-microgrid-standard/common/TimerHandler.h

```diff
@@ -0,0 +1,246 @@
+
+  TimerHandler(ACE_Reactor* reactor = nullptr)
+    : reactor_(reactor ? reactor : ACE_Reactor::instance())
+  {
+  }
+
+  virtual ~TimerHandler()
+  {
+    cancel_all();
+  }
+
+  template <typename EventType>
+  typename Timer<EventType>::MapPtr get_timers()
+  {
+    return static_cast<typename Timer<EventType>::MapPtr>(*this);
+  }
+
+  template <typename EventType>
+  typename Timer<EventType>::Ptr get_timer(const std::string& name = "")
+  {
+    auto& timers = *get_timers<EventType>();
+    const auto it = timers.find(name);
+    if (it == timers.end()) {
+      auto timer = std::make_shared<Timer<EventType>>();
+      timer->name = name;
+      timers[name] = timer;
+      return timer;
+    }
+    return it->second;
+  }
+
+  template <typename EventType>
+  void assert_inactive(typename Timer<EventType>::Ptr timer)
+  {
+    if (timer->active()) {
+      throw std::runtime_error("Can't schedule, timer already active!");
+    }
+  }
+
+  template <typename EventType>
+  void schedule(typename Timer<EventType>::Ptr timer)
+  {
+    Guard g(lock_);
+    assert_inactive<EventType>(timer);
+    const TimerId id = reactor_->schedule_timer(
+      this, &timer->id, ACE_Time_Value(timer->delay), ACE_Time_Value(timer->period));
+    timer->id = id;
+    active_timers_[id] = timer;
+  }
+
+  template <typename EventType>
+  void schedule(const std::string& name, EventType arg, Sec period, Sec delay = Sec(0))
+  {
+    Guard g(lock_);
+    auto timer = this->get_timer<EventType>(name);
+    assert_inactive<EventType>(timer);
+    timer->arg = arg;
+    timer->period = period;
+    timer->delay = delay;
+    schedule<EventType>(timer);
+  }
+
+  template <typename EventType>
+  void schedule_once(const std::string& name , EventType arg, Sec delay)
+  {
+    schedule(name, arg, Sec(0), delay);
+  }
+
+  template <typename EventType>
+  void schedule(EventType arg, Sec period, Sec delay = Sec(0))
+  {
+    schedule("", arg, period, delay);
+  }
+
+  template <typename EventType>
+  void schedule_once(EventType arg, Sec delay)
+  {
+    schedule(arg, Sec(0), delay);
+  }
+
+  template <typename EventType>
+  void timer_wont_run(typename Timer<EventType>::Ptr timer)
+  {
+    active_timers_.erase(timer->id);
+    timer->id = null_timer_id;
+  }
+
+  template <typename EventType>
+  void cancel(typename Timer<EventType>::Ptr timer)
+  {
+    if (!timer->active()) {
+      return;
+    }
+    reactor_->cancel_timer(timer->id);
+    timer_wont_run<EventType>(timer);
+  }
+
+  template <typename EventType>
+  void cancel(const std::string& name = "")
+  {
+    Guard g(lock_);
+    cancel<EventType>(this->get_timer<EventType>(name));
+  }
+
+  template <typename EventType>
+  void reschedule(const std::string& name, EventType arg)
+  {
+    Guard g(lock_);
+    auto timer = this->get_timer<EventType>(name);
+    cancel<EventType>(timer);
+    timer->arg = arg;
+    schedule(name, timer);
```

Comment:
I haven't seen the definition of this function in this class. Is it defined somewhere else?

Reply:
Hmmm, I think this function and the one below it became unused at some point, because I don't think there is a `schedule` that compile with this. I'll remove them.